ion/src/simulator/shared/main_sdl.cpp

```diff
@@ -41,40 +72,43 @@ namespace Main {
 
 static SDL_Window * sWindow = nullptr;
 static SDL_Renderer * sRenderer = nullptr;
-#if !EPSILON_SDL_SCREEN_ONLY
 static SDL_Texture * sBackgroundTexture = nullptr;
-#endif
 static bool sNeedsRefresh = false;
-#if EPSILON_SDL_SCREEN_ONLY
 static SDL_Rect sScreenRect;
-#endif
 
 void init() {
   if (SDL_Init(SDL_INIT_VIDEO) != 0) {
     SDL_Log("Could not init video");
     return;
   }
 
-  sWindow = SDL_CreateWindow(
-    "Epsilon",
-    SDL_WINDOWPOS_CENTERED,
-    SDL_WINDOWPOS_CENTERED,
+  uint32_t sdl_window_args = SDL_WINDOW_ALLOW_HIGHDPI | (argument_unresizable ? 0 : SDL_WINDOW_RESIZABLE);
+
+  if (argument_fullscreen) {
+    sdl_window_args = SDL_WINDOW_ALLOW_HIGHDPI | SDL_WINDOW_FULLSCREEN;
+  }
+
+  if (argument_screen_only) {
+    sWindow = SDL_CreateWindow(
+      "Epsilon",
+      SDL_WINDOWPOS_CENTERED,
+      SDL_WINDOWPOS_CENTERED,
+      Ion::Display::Width, Ion::Display::Height,
 #if EPSILON_SDL_SCREEN_ONLY
```

Comment:
It is bit hard to follow what is happening there with `sdl_window_args` now. Can keep one `SDL_CreateWindow` call, but adjust `sdl_window_args` and `sdl_window_init_height/width` (some new var) instead?